stdlib/list.gr

```diff
@@ -329,3 +329,54 @@ let join = (separator: String, items: List<String>) => {
     Some(s) => s,
   }
 }
+
+/**
+ * Reverses the list and appends items to the end
```

Comment:
```suggestion
 * Reverses the first list and appends the second list to the end.
```

@phated what do you think about mentioning that it's for efficiency? OCaml describes it as
```
`rev_append l1 l2` reverses `l1` and concatenates it with `l2`. This is equivalent to `(List.rev l1) @ l2`, but `rev_append` is tail-recursive and more efficient.
```